src/components/common/Stepper.vue

```diff
@@ -91,11 +94,22 @@ export default class Stepper extends Vue {
 
       case RouteNames.REGISTRATION_DEFINE_BUSINESS: return this.getRegistration.defineBusinessValid
       case RouteNames.REGISTRATION_PEOPLE_ROLES: return this.isAddPeopleAndRolesValid
-      case RouteNames.REGISTRATION_REVIEW_CONFIRM: return this.isRegistrationValid
+      case RouteNames.REGISTRATION_REVIEW_CONFIRM: return this.isRegistrationValid()
     }
     return false
   }
 
+  /** Check for define and people components are valid first
+  * Check for review component when the FileAndPay button is clicked */
+  private isRegistrationValid (): boolean {
+    if (this.isTypeFirm) {
+      if (!this.isPartialRegistrationValid) return false
+      if (!this.getClickFileAndPay) return this.isPartialRegistrationValid
+      return this.isFinalRegistrationValid
+    }
+    return this.isFinalRegistrationValid
+  }
+
```

Comment:
The logic in this method is the following:
1. Checks if it is a firm type or not
2. If two previous steppers are not valid, then return false; otherwise, move on to the next step.
3. If it's the first time clicking the button "File and Pay", then check "isPartialRegistrationValid".
4. If it is a firm type, then check "isFinalRegistrationValid".

Reply:
I think there are getters in create UI which already handle the validation before and after file and pay has been clicked https://github.com/bcgov/business-create-ui/blob/c0ab60baa549c77d87d5a29482c8d6a7f6b120b7/src/store/getters/state-getters.ts#L556 and https://github.com/bcgov/business-create-ui/blob/c0ab60baa549c77d87d5a29482c8d6a7f6b120b7/src/store/getters/state-getters.ts#L564

Reply:
Okay I see what you mean.
Have you checkout that this Component works in other places?
I know I have had to fix this exact error in this component for BEN dissolutions previously.
I think it would be worth it to check that this stepper function works the same across dissolutions, registrations and incorporations. 

If you have already checked it then I am happy with it.

Reply:
Because the ticket only shows the image of registration, I'm not sure whether this method is used for all filing types. If it is used for, then there will be a small tweak in the implementation. We will have to confirm with Yui later.